### PR TITLE
[IMP] web: display full tracebacks for error chains 

### PR DESCRIPTION
**Pull request by sdegueldre**
Since owl now uses error chains/causes when errors happen in the owl
lifecycle, when an error happens in the owl lifecycle, the displayed
tracebacks generally only contain the stack trace of where owl called
the corresponding lifecycle hook.

This commit modifies the error service and the error utils so that now,
when completing/annotating a traceback, we also add the tracebacks
(annotated when appropriate) of the error cause chain, as it contains
valuable debugging information.

This commit also makes it so that the QUnit suite logs the source of
each test failure (which may be an error with a chained stack trace)

### CODE REVIEW

File: addons/web/static/lib/owl/owl.js [hunk 1]
```diff
@@ -135,10 +135,10 @@
             console.warn(`[Owl] Unhandled error. Destroying the root component`);
             try {
                 node.app.destroy();
-            }
-            catch (e) {
+            } catch (e) {
                 console.error(e);
             }
+            throw error;
```
Comment (sdegueldre) [inline addons/web/static/lib/owl/owl.js:141]: needs to be done in owl

File: addons/web/static/src/core/errors/error_service.js [hunk 2]
```diff
@@ -61,16 +61,17 @@ function combineErrorNames(uncaughtError, originalError) {
  * @param {Error} originalError
  * @returns {Promise<void>}
  */
-async function completeUncaughtError(env, uncaughtError, originalError) {
+export async function completeUncaughtError(env, uncaughtError, originalError) {
```
Comment (ged-odoo) [inline addons/web/static/src/core/errors/error_service.js:64]: maybe should be moved to error_utils now?

Comment (ged-odoo) [reply addons/web/static/src/core/errors/error_service.js:64]: also, its api looks bad, but I am not inspired to give a better signature

File: addons/web/static/tests/qunit.js [hunk 3]
```diff
@@ -510,4 +516,44 @@ export function setupQUnit() {
...
+        const uncaughtError = new UncaughtPromiseError();
+        uncaughtError.unhandledRejectionEvent = ev;
+        if (originalError instanceof Error) {
+            originalError.errorEvent = ev;
+            await completeUncaughtError({}, uncaughtError, originalError);
+        }
+        originalError.stack = uncaughtError.traceback;
+        onUnhandledRejection(originalError);
+        // Do not log to the console as this will kill python test early
+        ev.preventDefault();
```
Comment (sdegueldre) [inline addons/web/static/tests/qunit.js:557]: reùpve

File: addons/web/static/tests/qunit.js [hunk 4]
```diff
@@ -398,8 +406,12 @@ export function setupQUnit() {
     QUnit.begin(function () {
         if (odoo.__DEBUG__.services["@web/core/errors/error_utils"]) {
             const errorUtils = odoo.__DEBUG__.services["@web/core/errors/error_utils"];
```
Comment (ged-odoo) [inline addons/web/static/tests/qunit.js:408]: i guess there is a reason for that, but why is this not imported at the top of the file, with an `import`, like everything else?

**Comment by ged-odoo** [comment]
robodoo r+

**Pull request closed without merging**